srcpkgs/libopenmpt/template

```diff
@@ -0,0 +1,85 @@
+# Template file for 'libopenmpt'
+pkgname=libopenmpt
+version=0.4.11
+revision=1
+wrksrc="libopenmpt-${version}+release.autotools"
+build_style=gnu-configure
+configure_args="--prefix=/usr
+	$(vopt_with zlib)
+	$(vopt_with mpg123)
+	$(vopt_with ogg) $(vopt_with vorbis) $(vopt_with vorbis vorbisfile)
+	$(vopt_with pulseaudio)
+	$(vopt_with portaudio) $(vopt_with portaudiocpp)
+	$(vopt_with sdl2) $(vopt_with sdl)
+	$(vopt_with sndfile)
+	$(vopt_with flac)
+	$(vopt_enable libopenmpt_modplug) $(vopt_enable libmodplug)
+	$(vopt_enable openmpt123)"
+hostmakedepends="pkg-config"
+makedepends="
+	$(vopt_if zlib zlib-devel)
+	$(vopt_if mpg123 mpg123-devel)
+	$(vopt_if ogg libogg-devel) $(vopt_if vorbis libvorbis-devel)
+	$(vopt_if pulseaudio pulseaudio-devel)
+	$(vopt_if portaudio portaudio-devel) $(vopt_if portaudiocpp portaudio-cpp-devel)
+	$(vopt_if sdl2 SDL2-devel) $(vopt_if sdl SDL-devel)
+	$(vopt_if sndfile libsndfile-devel)
+	$(vopt_if flac libflac-devel)"
+depends="
+	$(vopt_if zlib zlib)
+	$(vopt_if mpg123 libmpg123)
+	$(vopt_if ogg libogg) $(vopt_if vorbis libvorbis)
+	$(vopt_if pulseaudio libpulseaudio)
+	$(vopt_if portaudio portaudio) $(vopt_if portaudiocpp portaudio-cpp)
+	$(vopt_if sdl2 SDL2) $(vopt_if sdl SDL)
+	$(vopt_if sndfile libsndfile)
+	$(vopt_if flac libflac)"
+short_desc="Cross-platform C & C++ library to decode tracked music files (modules)"
+maintainer="a dinosaur <nick@a-dinosaur.com>"
+license="BSD-3-Clause"
+homepage="https://lib.openmpt.org/libopenmpt/"
+distfiles="https://lib.openmpt.org/files/libopenmpt/src/libopenmpt-${version}+release.autotools.tar.gz"
+checksum=260e92cc2f6af37113442bff2c75a3c36a09eba4078dc593203a0502f95d26bd
+conflicts="$(vopt_if libmodplug libmodplug)"
+
+post_install() {
+	vlicense LICENSE
+}
+
+# Package build options
+build_options="
```

Comment:
I realized that this might be a philosophical difference between distributions. Unlike Gentoo we do not want to provide every possible way to build a package in our package templates. Our build options are intended to be used for cases when it is technically necessary because a certain option can only be used on a subset of supported architectures or we think there are users who might want to built the package with or without a specific option.

Reply:
No I think you're correct and I'm being too overzealous on the options,
I misunderstood how the dependencies are being used and how xbps
works them out.
They are actually all entirely optional, though disabling MP3 and Vorbis
would disable MO3 support (not sure what zlib is used for but in
hindsight is silly to have an option for).

I agree with keeping pulseaudio an option, openmpt123 actually uses
pulse as the default output driver and with it enabled, requires
manually specifying a different driver on systems that don't run pulse
like my own.

SDL1.2 and SDL2 support is mutually exclusive and probably deserves an
option, but maybe keep them off by default because X11 dependency?

Portaudio is probably okay as non optional for openmpt123, its flakiness
on some setups seems like a good reason to keep SDL1/2 as an option.

The libmodplug shim will conflict with the package of the same name
and is *not* complete as libmodplug exposes a lot of internal
implementation details and should absolutely be off by default,
but may still be useful for those who want an easy way to substitute
the unmaintained libmodplug's aging and buggy replay engine in
software that still uses it.
instead of a build option perhaps it could be a separate package like
Debian does? (should the non destructive `libopenmpt_modplug` be
rolled into that too?)

Reply:
Cut a lot of the options, leaving open for comment.

Reply:
Looks alright for me. I'm not familiar with the libmodplud situation so I can't really comment on that. We could leave it disable for the time being and make changes later. If you think it is beneficial to avoid X11 dependencies, disabling SDL by default is OK.

Reply:
Alright, marking as resolved.